### PR TITLE
[CP Staging] fix: correctly pass transactions and reportActions to LHN

### PR DESCRIPTION
**Pull request by TMisiukiewicz**
### Details

Fixing regressions from #30242. In case no parentActions or parentReportActions were found, it was returning null instead of empty object. Additionally transactions were not passed to the component correctly, causing crashes when trying to access transaction details. 

I think none of these regressions would happen if we already had Typescript in this file, really happy we are making progress in this area and seeing more and more stuff rewritten 🙌 
### Fixed Issues

$ https://github.com/Expensify/App/issues/30998
PROPOSAL: 

### Tests

1. Send a picture through a chat
2. Open the picture
3. Refresh the page
4. Verify it refreshed correctly with no crash
- [x] Verify that no errors appear in the JS console

### QA Steps

1. Send a picture through a chat
2. Open the picture
3. Refresh the page
4. Verify it refreshed correctly with no crash
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/13985840/1076037d-4a1d-472e-b395-f70f63277fb1

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/13985840/ffde2921-8d7f-462d-9799-5543a928e702

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/13985840/6a017458-c1c3-422b-9261-6dc8266e3d84

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/13985840/62d45772-0691-4941-8e9e-f440d8d185f7

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/13985840/99821c5e-3cda-46fc-9bd6-97ee4d53fc5f

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/13985840/4ec8b59b-1d54-4956-9e15-2c855e62d554

</details>

### CODE REVIEW

File: src/components/LHNOptionsList/OptionRowLHNData.js [hunk 1]
```diff
@@ -71,7 +67,7 @@ function OptionRowLHNData({
     policy,
     receiptTransactions,
     parentReportActions,
-    transactionID,
+    transaction,
```
Comment (situchan) [inline src/components/LHNOptionsList/OptionRowLHNData.js:70]: `transaction` is missing propTypes and defaultProps

File: src/components/LHNOptionsList/LHNOptionsList.js [hunk 2]
```diff
@@ -60,12 +61,7 @@ const propTypes = {
     personalDetails: PropTypes.objectOf(participantPropTypes),
 
     /** The transaction from the parent report action */
-    transactions: PropTypes.objectOf(
-        PropTypes.shape({
-            /** The ID of the transaction */
-            transactionID: PropTypes.string,
-        }),
-    ),
+    transactions: transactionPropTypes,
```
Comment (situchan) [inline src/components/LHNOptionsList/LHNOptionsList.js:64]: This is not correct. `transactions` is object

File: src/components/LHNOptionsList/LHNOptionsList.js [hunk 3]
```diff
@@ -51,7 +52,7 @@ const propTypes = {
     reports: PropTypes.objectOf(reportPropTypes),
 
     /** Array of report actions for this report */
-    reportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+    reportActions: PropTypes.shape(reportActionPropTypes),
```
Comment (situchan) [inline src/components/LHNOptionsList/LHNOptionsList.js:55]: This is not correct. `reportActions` is object.

**Review by situchan — CHANGES_REQUESTED** [review]

**Comment by situchan** [comment]
Also, pull main when ready for review

File: src/components/LHNOptionsList/LHNOptionsList.js [hunk 4]
```diff
@@ -136,13 +132,10 @@ function LHNOptionsList({
...
-            const itemParentReportActions = reportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${itemFullReport.parentReportID}`];
-            const itemPolicy = policy[`${ONYXKEYS.COLLECTION.POLICY}${itemFullReport.policyID}`];
-            const itemTransaction = `${ONYXKEYS.COLLECTION.TRANSACTION}${lodashGet(
-                itemParentReportActions,
-                [itemFullReport.parentReportActionID, 'originalMessage', 'IOUTransactionID'],
-                '',
-            )}`;
+            const itemParentReportActions = reportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${itemFullReport.parentReportID}`] || {};
+            const itemPolicy = policy[`${ONYXKEYS.COLLECTION.POLICY}${itemFullReport.policyID}`] || {};
+            const itemTransactions = lodashGet(itemParentReportActions, [itemFullReport.parentReportActionID, 'originalMessage', 'IOUTransactionID'], '');
```
Comment (mountiny) [inline src/components/LHNOptionsList/LHNOptionsList.js:137]: This is always just a single transaction linked to the report action
```suggestion
            const itemTransaction = lodashGet(itemParentReportAction, [itemFullReport.parentReportActionID, 'originalMessage', 'IOUTransactionID'], '');
```

File: src/components/LHNOptionsList/LHNOptionsList.js [hunk 5]
```diff
@@ -136,13 +132,10 @@ function LHNOptionsList({
...
             const itemFullReport = reports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] || {};
             const itemReportActions = reportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`];
-            const itemParentReportActions = reportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${itemFullReport.parentReportID}`];
-            const itemPolicy = policy[`${ONYXKEYS.COLLECTION.POLICY}${itemFullReport.policyID}`];
-            const itemTransaction = `${ONYXKEYS.COLLECTION.TRANSACTION}${lodashGet(
-                itemParentReportActions,
-                [itemFullReport.parentReportActionID, 'originalMessage', 'IOUTransactionID'],
-                '',
-            )}`;
+            const itemParentReportActions = reportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${itemFullReport.parentReportID}`] || {};
```
Comment (mountiny) [inline src/components/LHNOptionsList/LHNOptionsList.js:135]: This is always just a single report action
```suggestion
            const itemParentReportAction = reportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${itemFullReport.parentReportID}`] || {};
```

File: src/components/LHNOptionsList/LHNOptionsList.js [hunk 6]
```diff
@@ -136,13 +132,10 @@ function LHNOptionsList({
...
+            const itemPolicy = policy[`${ONYXKEYS.COLLECTION.POLICY}${itemFullReport.policyID}`] || {};
+            const itemTransactions = lodashGet(itemParentReportActions, [itemFullReport.parentReportActionID, 'originalMessage', 'IOUTransactionID'], '');
+            const transactionsList = itemTransactions ? transactions[`${ONYXKEYS.COLLECTION.TRANSACTION}${itemTransactions}`] : transactions;
```
Comment (mountiny) [inline src/components/LHNOptionsList/LHNOptionsList.js:138]: I am not sure if we want to pass the entire collection of transactions in this case if itemTransaction would not be found
```suggestion
            const transactionsList = itemTransaction ? transactions[`${ONYXKEYS.COLLECTION.TRANSACTION}${itemTransaction}`] : transactions;
```

**Comment by TMisiukiewicz** [comment]
@situchan @mountiny thanks for your reviews. I addressed all the comments

File: src/components/LHNOptionsList/LHNOptionsList.js [hunk 7]
```diff
@@ -136,24 +132,21 @@ function LHNOptionsList({
...
             const itemComment = draftComments[`${ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT}${reportID}`] || '';
             const participantsPersonalDetails = OptionsListUtils.getPersonalDetailsForAccountIDs(itemFullReport.participantAccountIDs, personalDetails);
+
             return (
                 <OptionRowLHNData
                     reportID={reportID}
                     fullReport={itemFullReport}
                     reportActions={itemReportActions}
-                    parentReportActions={itemParentReportActions}
+                    parentReportActions={itemParentReportAction}
```
Comment (situchan) [inline src/components/LHNOptionsList/LHNOptionsList.js:146]: `parentReportActions` should also be changed in OptionRowLHNData

Comment (situchan) [reply src/components/LHNOptionsList/LHNOptionsList.js:146]: So remove this line and directly use `parentReportAction` from prop

https://github.com/Expensify/App/blob/4f9210729a368e16be965d1607a63559189a7e8c/src/components/LHNOptionsList/OptionRowLHNData.js#L78

**Comment by TMisiukiewicz** [comment]
@situchan thank you, could you re-review please when you have a moment?

File: src/components/LHNOptionsList/OptionRowLHNData.js [hunk 8]
```diff
@@ -36,9 +36,9 @@ const propTypes = {
     }),
 
     /** The actions from the parent report */
```
Comment (situchan) [inline src/components/LHNOptionsList/OptionRowLHNData.js:38]:
```suggestion
    /** The action from the parent report */
```

Comment (TMisiukiewicz) [reply src/components/LHNOptionsList/OptionRowLHNData.js]: good eye! done ✅

File: src/components/LHNOptionsList/LHNOptionsList.js [hunk 9]
```diff
@@ -136,24 +132,22 @@ function LHNOptionsList({
...
-            const itemPolicy = policy[`${ONYXKEYS.COLLECTION.POLICY}${itemFullReport.policyID}`];
-            const itemTransaction = `${ONYXKEYS.COLLECTION.TRANSACTION}${lodashGet(
-                itemParentReportActions,
-                [itemFullReport.parentReportActionID, 'originalMessage', 'IOUTransactionID'],
-                '',
-            )}`;
+            const itemParentReportActions = reportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${itemFullReport.parentReportID}`] || {};
+            const itemParentReportAction = itemParentReportActions[itemFullReport.parentReportActionID] || {};
+            const itemPolicy = policy[`${ONYXKEYS.COLLECTION.POLICY}${itemFullReport.policyID}`] || {};
+            const transactionID = lodashGet(itemParentReportAction, [itemFullReport.parentReportActionID, 'originalMessage', 'IOUTransactionID'], '');
```
Comment (situchan) [inline src/components/LHNOptionsList/LHNOptionsList.js:138]:
```suggestion
            const transactionID = lodashGet(itemParentReportAction, ['originalMessage', 'IOUTransactionID'], '');
```

Had to watch this with 🦅's eye 😄

**Review by situchan — CHANGES_REQUESTED** [review]

**Comment by situchan** [comment]
Do you think `transactionID` is enough to satisfy this comment? I am concerned about removing `transaction` from dependency

Screenshot 2023-11-08

File: src/components/LHNOptionsList/LHNOptionsList.js (hunk 9 above)
Comment (TMisiukiewicz) [reply src/components/LHNOptionsList/LHNOptionsList.js:138]: oh wow, the 🦅 eye indeed 🤯

**Comment by TMisiukiewicz** [comment]
@situchan good point, I'd say it's safer to bring back whole transaction as a dependency. I reverted changes to use `transaction`.

**Comment by situchan** [comment]
ok scan transaction title is updated in LHN

https://github.com/Expensify/App/assets/108292595/38a6071b-5d7a-4e66-ae31-8e7388cc64db

**Comment by situchan** [comment]
Console errors from bad Onyx data. Not sure if it's coming from backend or Onyx but out of scope anyway

Simulator Screenshot


Simulator Screenshot

**Review by situchan — APPROVED** [review]

**Review by mountiny — APPROVED** [review]
Thanks for the fix

**Comment by mountiny** [comment]
The failing reasssure test was because of different changes in main, none of the changes in this PR influence the ReportScreens performance

**Pull request merged**